Ical.Net/CalendarComponents/RecurringComponent.cs

```diff
@@ -113,9 +112,30 @@ public virtual IList<RecurrencePattern> RecurrenceRules
         set => Properties.Set("RRULE", value);
     }
 
+    /// <summary>
+    /// Gets or sets the recurrence identifier for a specific instance of a recurring event.
+    /// </summary>
+    /// <remarks>Use <see cref="RecurrenceInstance"/> instead, which
+    /// supports the RANGE parameter for recurring events.</remarks>
+    [Obsolete("Use RecurrenceInstance instead, which supports the RANGE parameter.")]
     public virtual CalDateTime? RecurrenceId
     {
-        get => Properties.Get<CalDateTime>("RECURRENCE-ID");
+        get => RecurrenceInstance?.Range == RecurrenceRange.ThisInstance ? RecurrenceInstance.StartTime : null;
+        set => RecurrenceInstance = value is null ? null : new RecurrenceId(value, RecurrenceRange.ThisInstance);
+    }
+
+    /// <summary>
+    /// Gets or sets the recurrence identifier for a specific instance of a recurring event.
+    /// <para/>
+    /// The <see cref="RecurrenceId.Range"/> sets the scope of the recurrence instance:
+    /// With <see cref="RecurrenceRange.ThisInstance"/>, the instance is limited to the specific
+    /// occurrence identified by the <see cref="RecurrenceId.StartTime"/>.<br/>
+    /// With <see cref="RecurrenceRange.ThisAndFuture"/>, the instance applies to the specified
+    /// <see cref="RecurrenceId.StartTime"/> and all future occurrences.
+    /// </summary>
+    public virtual RecurrenceId? RecurrenceInstance
```

Comment:
I feel the name `RecurrenceInstance` is misleading. Not sure, what a good name would be (other than `RecurrenceId`). Maybe `RecurrenceIdentifier`? Somewhat long, but that's close to how it is named in the rfc.